addons/stock/models/stock_quant.py

```diff
@@ -327,6 +330,18 @@ def _onchange_location_or_product_id(self):
         if vals:
             self.update(vals)
 
+    @api.onchange('inventory_quantity')
+    def _onchange_inventory_quantity(self):
+        if self.location_id and self.location_id.usage == 'inventory':
+            warning = {
+                'title': _('You cannot modify inventory loss quantity'),
+                'message': _(
+                    'This type of quant balances your inventory quantity.\n'
+                    'Instead of modify it directly, please modify your real quant.'
+                )
```

Comment:
not sure the user should be faced with the 'quant' wording, maybe ask lap

Reply:
Change the message to:

"Editing quantities in an Inventory Adjustment location is forbidden, those locations are used as counterpart when correcting the quantities."